sBTC/clarity-vm-tsd.md

```diff
@@ -0,0 +1,84 @@
+
+By improving the Clarity VM, the blockchain will be able to handle more transactions quickly, making it easier for people to use and enjoy the benefits of the Stacks blockchain. This upgrade will help make the blockchain more reliable and user-friendly.
+
+## Problem Statement:
+
+> 👉 _What specific problem does Clarity VM initiative solve?_
+
+</br>
+
+The Clarity VM in the stacks-node faces inefficiencies in transaction processing, limiting the number of transactions that can be included in a block. This hampers user experience and transaction capacity. An improved VM is essential to address these limitations.
+
+With an improved Clarity VM, transactions can be processed much faster. This enhancement will enable an increase in the number of transactions that can fit into each block and improve the overall user experience significantly.
+
+## Solution:
+
+> 👉 _How does the Clarity VM initiative provide a solution to the identified problem?_
+
+</br>
+
+The plan to speed up the Clarity VM is to compile Clarity contracts into [WebAssembly (Wasm)](https://webassembly.org/), and then integrate a Wasm runtime into the Clarity VM. Wasm has proven to be an effective tool for fast, safe, and portable code execution, and is being used in several other blockchain projects. Initial back-of-the-envelope testing shows that a Wasm runtime has the potential to be at least an order of magnitude faster, assuming we do not hit other bottlenecks, e.g. MARF reads/writes.
+
+## Technical Requirements:
+
+> 👉 _What technical requirements and criteria must the Clarity VM initiative meet to be fully functional and successfully implemented?_
+
+</br>
+
+The new compiler and runtime must execute all valid Clarity code, including the code currently supported by the existing runtime. If any changes are made to Clarity to simplify or improve the implementation, this runtime will be epoch gated, and the old runtime will be used before its activation height.
+
+The stacks-node should efficiently store the compiled Wasm code alongside the Clarity source for all published contracts and retrieve it quickly when a contract is called. The saved Wasm should be accessible via RPC calls for verification and debugging purposes.
+
+The runtime needs to establish a mechanism to interact with the MARF using Clarity's native functions for reading/writing data-vars and maps.
+
+Furthermore, the runtime must ensure all the isolation, security, and predictability provided by the current runtime. Modifying state outside of well-defined interfaces should be impossible, and contract execution must be predictable and consistent in its execution time.
+
+## Architectural Diagrams:
+
+Clarity → clar2wasm (uses walrus to build Wasm) → Clarity VM (uses wasmtime)
+
+```
+			   +--------+   clar2wasm    +-------------------+
+               |        | -------------> |                   |
```

Comment:
@wileyj the ascii diagram is out of alignment.  perhaps we should just remove it for now - and add a JPEG or something - when Brice returns.

Reply:
![Screenshot 2023-07-27 at 17 09 40](https://github.com/stacks-network/stacks/assets/2847772/25733888-03b9-4c4d-8503-05324c36065d)

looks fine for me, i think you have an outdated view for the PR: https://github.com/stacks-network/stacks/blob/chore/sbtc-docs/sBTC/clarity_vm.md

Reply:
looks very neat